server/datastore/mysql/hosts.go

```diff
@@ -5139,8 +5169,8 @@ func (ds *Datastore) loadHostLite(ctx context.Context, id *uint, identifier *str
     SELECT
       h.id,
       h.team_id,
-      h.osquery_host_id,
-      h.node_key,
+      COALESCE(h.osquery_host_id, '') AS osquery_host_id,
+      COALESCE(h.node_key, '') AS node_key,
```

Comment:
Is this relevant to the fix? I think we do support and expect that those fields can be nil?

Reply:
@mna, I ran into this issue in the `runPostDEPEnrollment` [worker](https://github.com/fleetdm/fleet/pull/20162/files#diff-5792cd76cf90483f0a4d8d1424588dad244b733ce55fed1882ca0738b49123ccR173), which calls `HostLiteByIdentifier` as part of `SetOrUpdateHostEmailsFromMDMIdPAccountsByHostUUID`. At this point in the process, we have a `hosts` entry that was created when the device was ingested from DEP sync, but it doesn't have a `node_key` or `orbit_node_key` so `HostLite` errors without the `COALESCE`:
```
Scan error on column index 3, name \"node_key\": converting NULL to string is unsupported
```

I'm totally open to alternatives here if you think there's a better approach. I considered changing the `HostLite` struct to use string pointers for these nullable values. I also considered writing a special method that just gets the host ID by the UUID since that's the only reason I need to call `HostLiteByIdentifier`. WDYT?

Reply:
Oh my bad, I thought the target struct already had pointers to support those potentially-NULL columns. It must be because we never call that HostLite in this situation then. I'm fine with the fix you implemented, was just curious as it seemed unrelated.